plugins/module_utils/common.py

```diff
@@ -921,23 +922,27 @@ def get_legacy_docker_diffs(self):
         return result
 
 
-def clean_dict_booleans_for_docker_api(data):
+def clean_dict_booleans_for_docker_api(data, allow_sequences=False):
     '''
     Go doesn't like Python booleans 'True' or 'False', while Ansible is just
     fine with them in YAML. As such, they need to be converted in cases where
     we pass dictionaries to the Docker API (e.g. docker_network's
     driver_options and docker_prune's filters).
     '''
+    def sanitize_value(value):
+        if allow_sequences and is_sequence(value):
+            return type(value)(sanitize_value(v) for v in value)
```

Comment:
The API won't accept anything else than lists anyway, so it doesn't make to preserve the exact sequence type:
```suggestion
            return [sanitize_value(v) for v in value]
```

Reply:
Another thing is that this code now processes lists recursively. So `[['a']]` will be kept, instead of being converted to `["['a']"]`. (The latter is not what the user wants, but what the API wants :) )

Reply:
Yeah, now that I look at the [API](https://github.com/docker/docker-py/blob/96c12726fdb9b24e985b4de74d5c82145ccd8185/docker/utils/utils.py#L366-L377)  they just convert every list item to strings so no recursion makes sense. I'll update this and the docs later on today.